### ISSUE TITLE
tsne plot for keras tokenizer word emebdding for 5 classes of texts

### ISSUE THREAD

**Issue by vinayakumarr** [fchollet/keras#6326]
I have 5 class text classification and I used keras tokeinizer with LSTM. 

tk = keras.preprocessing.text.Tokenizer(nb_words=20000, lower=True, split=" ")
tk.fit_on_texts(x)

x = tk.texts_to_sequences(x)

tk = keras.preprocessing.text.Tokenizer(nb_words=20000, lower=True, split=" ")
tk.fit_on_texts(t)
t = tk.texts_to_sequences(t)

y = np.array(y)

max_len = 500
print "max_len ", max_len
print('Pad sequences (samples x time)')

x = sequence.pad_sequences(x, maxlen=max_len)
t = sequence.pad_sequences(t, maxlen=max_len)
y= to_categorical(y)

max_features = 20000

model = Sequential()
print('Build model...')

model = Sequential()
model.add(Embedding(max_features, 128, input_length=max_len, dropout=0.1))
model.add(LSTM(128, dropout_W=0.1, dropout_U=0.1))
model.add(Dense(5))
model.add(Activation('softmax'))

After training, I have made an attempt to plot the weights in tsne 

def plot_embeddings(embeddings, names):
    model = TSNE(n_components=2, random_state=0)
    vectors = model.fit_transform(embeddings)
    x, y = vectors[:, 0], vectors[:, 1]
    print("vinay")
    print(x.shape)
    print(t.shape)
    fig, ax = pyplot.subplots(figsize=(5,4))
    ax.scatter(x, y)
    ax.tick_params(direction='in')
    
    for i, tname in enumerate(names):
        ax.annotate(x[i], y[i])
    pyplot.show()

model.load_weights("checkpoint-35.hdf5")
embeddings = model.layers[0].get_weights()[0]
names = list(vocabulary_names)
plot_embeddings(embeddings, names)

But I have 5 classes, So how to group the data in tsne. Now i am getting a plot with all points. Could you please tell how to group these data when i use keras tokenizer with word emebdding
